### ISSUE TITLE
Dart allow declare a typedef with the type parameter which is not well bounded.

### ISSUE THREAD

**Issue by iarkh** [dart-lang/sdk#36869]
Dart SDK Version: 2.3.0-dev.0.5
OS: Windows 10 64 bit

Dart allow declare a typedef with the type parameter which is not well bounded.

**Comment by eernstg** [comment]
I agree: The raw type `A` is used as a bound in the body of `AAlias`, but `A` does not have simple bounds (`T` has a bound `A<T>`, and that bound contains a type parameter declared by `A`), so it's a compile-time error to use a raw `A` as a bound of any type parameter, cf. [this](https://github.com/dart-lang/language/blob/87bd63679a8233776f559d7ab4ed45a49456f5bb/specification/dartLangSpec.tex#L5568) location in the spec.

The error is actually that we're using a raw type with non-simple bounds as a bound, not that any type in `Function<Y extends A> ()` is not well-bounded. However, the error message from the analyzer is misleading: It sounds like _every_ raw type used as a bound is an error, but the error only exists when the bound is/contains a type which is raw and which has non-simple bounds.

The common front end does not detect the error. I believe this amounts to a special case of #33184, and it makes sense to create this new issue for that special case. I've adjusted the labels accordingly.
